js_kadai4.js

```diff
@@ -0,0 +1,176 @@
+let quiz;
+let currentQuiz = 0;
+let correctNum = 0;
+
+document.addEventListener('DOMContentLoaded', () => {
+    document.getElementById('btn_start').addEventListener('click', () => {
+
+        const title = document.getElementById('title');
+        const explain = document.getElementById('explain');
+        title.textContent = '取得中';
+        explain.textContent = '少々お待ちください';
+        
+        const btn_start = document.getElementById('btn_start');
+        btn_start.remove();
+
+        //問題を取得
+        fetch('https://opentdb.com/api.php?amount=10&encode=url3986')
+        .then(
+            response => {
+                if (response.ok) {
+                    return response.json();
+                } else {
+                    throw new Error();
+                }
+            }
+        )
+        .then(
+            data => {
+                if (data.response_code !== 0) {
+                    throw new Error();
+                }
+                console.log(data);
+                quiz = data;
+                showQuiz(currentQuiz);
+            }
+        )
+        .then(
+            error => {
+                console.log(error);
+            }
+        )
```

Comment:
```suggestion
        try {
            const response = await fetch('https://opentdb.com/api.php?amount=10&encode=url3986');
            if (response.ok) {
                quiz = await response.json();
                showQuiz(currentQuiz);
            } else {
                console.error(`Http status error. status code is ${response.status}, http headers ${response.headers}`)
            }
        } catch(e) {
            console.error(e);
        }
```

昔ながらの `Promise` を用いる書き方でも問題はないですが
`async` `await` を使った記述の方がすっきり書けますし、他の言語との差異も少なくなるのでおススメです。